compiler/rustc_middle/src/ty/relate.rs

```diff
@@ -523,9 +522,7 @@ pub fn super_relate_consts<R: TypeRelation<'tcx>>(
 
         (ty::ConstKind::Param(a_p), ty::ConstKind::Param(b_p)) => a_p.index == b_p.index,
         (ty::ConstKind::Placeholder(p1), ty::ConstKind::Placeholder(p2)) => p1 == p2,
-        (ty::ConstKind::Value(a_val), ty::ConstKind::Value(b_val)) => {
-            check_const_value_eq(relation, a_val, b_val, a, b)?
-        }
+        (ty::ConstKind::Value(a_val), ty::ConstKind::Value(b_val)) => a_val == b_val,
```

Comment:
This is the essence of this PR. There is no more special logic for relating constants, we can just compare them with the derived PartialEq impl.